app/models/Database.scala

```diff
@@ -0,0 +1,149 @@
+    ScreeningRoom(1))
+  screenigsList += Screening(5,
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(25).withHourOfDay(21).withMinuteOfHour(30),
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(25).withHourOfDay(23).withMinuteOfHour(30),
+    moviesList(5),
+    ScreeningRoom(3))
+  screenigsList += Screening(6,
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(26).withHourOfDay(21).withMinuteOfHour(0),
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(26).withHourOfDay(22).withMinuteOfHour(0),
+    moviesList(6),
+    ScreeningRoom(2))
+  screenigsList += Screening(8,
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(27).withHourOfDay(17).withMinuteOfHour(0),
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(27).withHourOfDay(19).withMinuteOfHour(0),
+    moviesList(1),
+    ScreeningRoom(1))
+  screenigsList += Screening(9,
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(28).withHourOfDay(17).withMinuteOfHour(0),
+    new DateTime().withYear(2022).withMonthOfYear(12).withDayOfMonth(28).withHourOfDay(19).withMinuteOfHour(0),
+    moviesList(2),
+    ScreeningRoom(2))
+
+  val reservationsList = new mutable.ListBuffer[Reservation]()
+
+
+  def validateReservation(reservation: Reservation): Boolean = {
+
+
+    var idCorrectness = false
+    for(screening <- screenigsList){
+      if(screening.id == reservation.screeningId){
+        idCorrectness = true
+      }
+    }
+
+    if(!idCorrectness) return false
+
+    val name = reservation.name.split(" ")
+    for(word <- name){
+      if(name.length != 2 || !word.charAt(0).isUpper || word.length<3) return  false
+    }
+
+    val surname = reservation.name.split(" ")(1).split("-")
+    if(surname.length==2){
+      for(word <- surname){
+        if(!word.charAt(0).isUpper) return false
+      }
+    } else if(surname.length>2){
+      return false
+    }
+
+
+    var seatsCorrectness = true
+
+    for (screening <- screenigsList) {
+
+      if (screening.id == reservation.screeningId) {
+
+        val reservationTime:DateTime = DateTime.now()
+        if(reservationTime + 15.minutes > screening.startDateTime) return false
+
+
+        val testScreeningRoom : ArrayBuffer[ArrayBuffer[Int]] = new ArrayBuffer[ArrayBuffer[Int]]
+        screening.screeningRoom.seats.foreach(testScreeningRoom += _.clone())
+
+        for(seatReservation <- reservation.seats){
+          if(!screening.screeningRoom.seats.flatten.contains(seatReservation.seatNumber)) seatsCorrectness = false
+          testScreeningRoom(seatReservation.seatNumber / 10)-=seatReservation.seatNumber
+        }
+
+        for(row <-testScreeningRoom){
+          for(seat <- row){
+            if(!row.contains(seat-1) && !row.contains(seat+1) && seat%10 >= 2)seatsCorrectness = false
+          }
+        }
+      }
+    }
+
+    if(!seatsCorrectness)return false
+
+    return true
```

Comment:
W jaki sposób użytkownik dowie się co poszło źle podczas jego rezerwacji? Podczas walidacji dość dobrą praktyką jest podać przyczynę błędu - jak przetestowałbyś tą funkcje by mieć pewność, że walidacja została odrzucona np. na podstawie czasu rezerwacji a jak, że odrzuciliśmy ją ponieważ wybrane zostały złe miejsca? W jaki sposób to odróżnisz?

Reply:
funkcja mogłaby zamiast booleana zwracać stringa z konkretnym błędem który uniemożliwia rezerwacje i pusty string kiedy rezerwacja by się powiodła